.github/workflows/sync-upstream-pr.yml

```diff
@@ -0,0 +1,136 @@
+name: Sync with Upstream (create)
+
+on:
+  schedule:
+    - cron: "0 0 1 * *" # Runs at 00:00 on the first day of every month
+  workflow_dispatch:
+  # TODO: Remove this trigger once the workflow is ready
+  pull_request:
+    branches:
+      - main
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.user.login != 'github-actions[bot]'
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+          ref: main
+
+      - name: Create new branch from main
+        run: |
+          git checkout -b update-from-upstream-${{ github.run_number }}
+
+      - name: Configure Git
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+
+      #TODO: Remove this step once the workflow is ready
+      - name: Merge the feature branch for the workflow
+        run: |
+          git merge --squash origin/gc/sync-upstream
+          git commit -m "Add sync-upstream.yml workflow (#7)"
+
+      - name: Pull changes from upstream and rebase
+        id: rebase
+        run: |
+          git remote add upstream https://github.com/actions/runner-images.git
+          git fetch upstream
+          echo "upstream-sha=$(git rev-parse upstream/main)" >> $GITHUB_OUTPUT
+          git rebase upstream/main || echo "conflicts=true" >> $GITHUB_OUTPUT
```

Comment:
By the way (starting a separate thread for this one), ... I had a _shower thought_ this morning, which is...

Is merging actually so bad? Maybe I am wrong about rebasing being the only realistic strategy...

I'm wondering if we merge the upstream commits _into_ our branch, what would go wrong?

1. You'd have to resolve the conflicts only once and they would be correct - no need for this squashing discussion.
2. `git blame` would work properly.
3. The history might not look the best: it would be merge commits mixed with non-merge commits which are our changes, and the GitHub UI doesn't show this very nicely so people would need to use `git log --graph` or another tool to look at it.
    - But **actually** maybe this is the right history after all: it would show merges when we merge - in other words, it would show what we actually did.
4. No need for force pushing. 

Just testing this locally with two merges and a fake commit, this is what the graph ends up looking like:

![image](https://github.com/grafana/runner-images/assets/321014/ab394e7b-7a42-4320-befc-df0cbbc183d8)

which isn't the easiest to read, but it is actually correct.

WDYT? sorry for bringing this back up ... 🙃

Reply:
That was my other option, when we make new features we squash and merge, but when merging the changes from upstream we create a merge commit, we let the workflow handle merging the upstream sync PR to be sure there are no mistakes.

I don't think that would be too bad, we would solve the same amount of conflicts and would keep the git history true to the timeline of commits.